plugins/module_utils/gitlab.py

```diff
@@ -119,8 +119,13 @@ def gitlab_authentication(module, min_version=None):
             resp_data = resp.json()
             gitlab_oauth_token = resp_data["access_token"]
 
-        gitlab_instance = gitlab.Gitlab(url=gitlab_url, ssl_verify=verify, private_token=gitlab_token,
-                                        oauth_token=gitlab_oauth_token, job_token=gitlab_job_token, api_version=4)
+        if gitlab.__version__ >= LooseVersion('3.8.0'):
+            gitlab_instance = gitlab.Gitlab(url=gitlab_url, ssl_verify=verify, private_token=gitlab_token,
+                                            oauth_token=gitlab_oauth_token, job_token=gitlab_job_token, api_version=4, keep_base_url=True)
+        else:
+            gitlab_instance = gitlab.Gitlab(url=gitlab_url, ssl_verify=verify, private_token=gitlab_token,
+                                            oauth_token=gitlab_oauth_token, job_token=gitlab_job_token, api_version=4)
+
```

Comment:
How about changing this to:
```suggestion
        kwargs = {
            'url': gitlab_url,
            'ssl_verify': verify,
            'private_token': gitlab_token,
            'oauth_token': gitlab_oauth_token,
            'job_token': gitlab_job_token,
            'api_version': 4,
        }
        if gitlab.__version__ >= LooseVersion('3.8.0'):
            kwargs['keep_base_url'] = True
        gitlab_instance = gitlab.Gitlab(**kwargs)

```
That way it's easier extendable when other version-dependent arguments are added.